classes/class-wpcom-liveblog-entry-embed.php

```diff
@@ -0,0 +1,296 @@
+<?php 
+
+/**
+ * Class WPCOM_Liveblog_Entry_Embed
+ *
+ * Store oembeds in comment meta instead of post meta.
+ */
+class WPCOM_Liveblog_Entry_Embed extends WP_Embed {
+
+	public function __construct() {
+		return; //nothing happens during __construct for now
+	}
+
+	/**
+	 * If a post/page was saved, then output JavaScript to make
+	 * an AJAX request that will call WP_Embed::cache_oembed().
+	 *
+	 * Override the default method in a way it does nothing as
+	 * we don't need to pre-populate the cache as comments are
+	 * being created on the fly on frontend
+	 */
+	public function maybe_run_ajax_cache() {
+		return;
+	}
+
+	/**
+	 * The {@link do_shortcode()} callback function.
+	 *
+	 * Attempts to convert a URL into embed HTML. Starts by checking the URL against the regex of the registered embed handlers.
+	 * If none of the regex matches and it's enabled, then the URL will be given to the {@link WP_oEmbed} class.
+	 *
+	 * @param array $attr {
+	 *     Shortcode attributes. Optional.
+	 *
+	 *     @type int $width  Width of the embed in pixels.
+	 *     @type int $height Height of the embed in pixels.
+	 * }
+	 * @param string $url The URL attempting to be embedded.
+	 * @param int $comment_id The Comment ID of currently processed comment
+	 * @return string|false The embed HTML on success, otherwise the original URL.
+	 *                      `->maybe_make_link()` can return false on failure.
+	 */
+	public function shortcode( $attr, $url = '', $comment = null ) {
+		if ( ! empty( $comment ) ) {
+			$comment = get_comment( $comment );
+		}
+
+		if ( empty( $comment ) ) {
+			$comment = get_comment();
+		}
+
+		if ( empty( $url ) && ! empty( $attr['src'] ) ) {
+			$url = $attr['src'];
+		}
+
+		$this->last_url = $url;
+
+		if ( empty( $url ) ) {
+			$this->last_attr = $attr;
+			return '';
+		}
+
+		$rawattr = $attr;
+		$attr = wp_parse_args( $attr, wp_embed_defaults( $url ) );
+
+		$this->last_attr = $attr;
+
+		// kses converts & into &amp; and we need to undo this
+		// See https://core.trac.wordpress.org/ticket/11311
+		$url = str_replace( '&amp;', '&', $url );
+
+		// Look for known internal handlers
+		$handlers = $this->handlers;
+		//check for handlers registered for wp_embed class using all the helper functions
+		if ( true === isset( $GLOBALS['wp_embed'] )
+			 && is_a( $GLOBALS['wp_embed'], 'WP_Embed' )
+			 && is_array( $GLOBALS['wp_embed']->handlers )
+		) {
+			//marge those in a single array
+			$handlers = array_replace_recursive( $GLOBALS['wp_embed']->handlers, $this->handlers );
+		}
+		ksort( $handlers );
+		foreach ( $handlers as $priority => $handlers ) {
```

Comment:
This is code taken from VIP.

Reply:
Where exactly? Still, doesn't mean it's perfect, so we should probably change it :)

Reply:
This whole class is from the wp com
helper, embeds where broken on non-vip installs as embed code was using post meta. This class looks to be a taken from core then updated to use comment meta instead of post meta

Reply:
That's fine, and may be the case, but I don't think we should be re-declaring the variable here. Whether it's from core or VIP-written, if there are improvements to be had we should make them now.

Reply:
No problem, we’re happy to make changes. The reason I didn’t alter any of it in the first place was I wasn’t sure if I’d be allowed too but happy to look for ways to improve it. Thanks